internal/pkg/keychain/keychain_darwin.go

```diff
@@ -0,0 +1,92 @@
+//go:build darwin
+
+package keychain
+
+import (
+	"strings"
+
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/log"
+	"github.com/confluentinc/cli/internal/pkg/netrc"
+	"github.com/keybase/go-keychain"
+)
+
+const (
+	accessGroup = "cli"
+	isCloud     = true
+	sep         = "?"
+)
+
+func Write(ctxName, url, username, password string) error {
+	service := netrc.GetLocalCredentialName(isCloud, ctxName)
+
+	item := keychain.NewGenericPassword(service, url, username+"-"+url, []byte(username+sep+password), accessGroup)
+	item.SetSynchronizable(keychain.SynchronizableNo)
+	item.SetAccessible(keychain.AccessibleWhenUnlocked)
+
+	err := keychain.AddItem(item)
+	if err == keychain.ErrorDuplicateItem {
```

Comment:
that's a lot more lines of code rather than `keychain.NewGenericPassword(` in one line. Looking up makes sense too but i think `duplicate item` error is doing the same thing, letting us know that the entry is already there